http_client.go

```diff
@@ -318,6 +319,14 @@ func (c *HTTPClient) Send(data []byte) (response []byte, err error) {
 		}
 	}
 
+	if (c.config.CloseConnectionOnResponseStatus > 0) {
```

Comment:
How about I'll make the default to close on `400`, but leave the config in if somebody wants it to use for another status code?